src/main.c

```diff
@@ -880,10 +880,48 @@ ubus_poe_reload_cb(struct ubus_context *ctx, struct ubus_object *obj,
 	return UBUS_STATUS_OK;
 }
 
+static const struct blobmsg_policy ubus_poe_manage_policy[] = {
+	{ "port", BLOBMSG_TYPE_STRING },
+	{ "enable", BLOBMSG_TYPE_BOOL },
+};
+
+static int ubus_poe_manage_cb(struct ubus_context *ctx,
+			      struct ubus_object *obj,
+			      struct ubus_request_data *req,
+			      const char *method, struct blob_attr *msg)
+{
+	struct blob_attr *tb[ARRAY_SIZE(ubus_poe_manage_policy)];
+	const struct port_config *port;
+	const char *port_name;
+	size_t i;
+
+	blobmsg_parse(ubus_poe_manage_policy,
+		      ARRAY_SIZE(ubus_poe_manage_policy),
+		      tb, blob_data(msg), blob_len(msg));
+	if (!tb[0] || !tb[1])
+		return UBUS_STATUS_INVALID_ARGUMENT;
+
+	port_name = blobmsg_get_string(tb[0]);
+	for (i = 0; i < config.port_count; i++) {
+		port = &config.ports[i];
+		if (!port->valid || strcmp(port_name, port->name))
+			continue;
+		if (!port->enable) {
+			ULOG_NOTE("Cannot manage disabled port '%s'\n", port_name);
+			return UBUS_STATUS_INVALID_ARGUMENT;
```

Comment:
Sounds like a noble effort, but the command fails with `Command failed: Invalid argument`, and the ubus caller doesn't see the NOTE message. So why bother? Replace `!port->valid` with `!port->enable` in the first predicate, and drop the second if statement altogether. 

Yes, it's safe to assume `!valid` implies `!enable`.

Reply:
Well, the main point wasn't the log message, but making a distinction between "port does not exist" and "port cannot be managed".  For example with "lan2" disabled in config, "lan3" disabled by ubus command, and all other ports enabled:
```
root@gs1900-10hp-f:~# ubus  call poe info
{
        "firmware": "v22.4",
        "mcu": "ST Micro ST32F100 Microcontroller",
        "budget": 77.000000,
        "consumption": 15.000000,
        "ports": {
                "lan1": {
                        "priority": 0,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan2": {
                        "priority": 0,
                        "mode": "PoE+",
                        "status": "Disabled"
                },
                "lan3": {
                        "priority": 0,
                        "mode": "PoE+",
                        "status": "Disabled"
                },
                "lan4": {
                        "priority": 0,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan5": {
                        "priority": 0,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan6": {
                        "priority": 0,
                        "mode": "PoE+",
                        "status": "Delivering power",
                        "consumption": 4.200000
                },
                "lan7": {
                        "priority": 0,
                        "mode": "PoE+",
                        "status": "Delivering power",
                        "consumption": 4.300000
                },
                "lan8": {
                        "priority": 0,
                        "mode": "PoE+",
                        "status": "Delivering power",
                        "consumption": 6.800000
                }
        }
}
root@gs1900-10hp-f:~# ubus  call poe manage '{"port":"lan2","enable":true}'
Command failed: Invalid argument
root@gs1900-10hp-f:~# ubus  call poe manage '{"port":"lan42","enable":true}'
Command failed: Not found
root@gs1900-10hp-f:~# ubus  call poe manage '{"port":"lan3","enable":true}'
```
Of course, if you insist, we could just return UBUS_STATUS_INVALID_ARGUMENT for all errors.  Then I agree that the "valid" test can be replaced by the "enable" test.